ci/docker/integration.dockerfile

```
@@ -0,0 +1,58 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+FROM apache/arrow-dev:amd64-conda-integration
+
+ENV ARROW_USE_CCACHE=OFF
+ENV ARROW_CPP_EXE_PATH=/build/cpp/debug
+ENV ARROW_NANOARROW_PATH=/build/nanoarrow
+ENV ARROW_RUST_EXE_PATH=/build/rust/debug
+ENV BUILD_DOCS_CPP=OFF
+ENV ARROW_INTEGRATION_CPP=ON
+ENV ARROW_INTEGRATION_CSHARP=ON
+ENV ARROW_INTEGRATION_GO=ON
+ENV ARROW_INTEGRATION_JAVA=ON
+ENV ARROW_INTEGRATION_JS=ON
+ENV ARCHERY_INTEGRATION_WITH_NANOARROW="1"
+ENV ARCHERY_INTEGRATION_WITH_RUST="1"
```

```suggestion
ENV ARROW_USE_CCACHE=OFF \
        ARROW_CPP_EXE_PATH=/build/cpp/debug \
        ARROW_NANOARROW_PATH=/build/nanoarrow \
        ARROW_RUST_EXE_PATH=/build/rust/debug \
        BUILD_DOCS_CPP=OFF \
        ARROW_INTEGRATION_CPP=ON \
        ARROW_INTEGRATION_CSHARP=ON \
        ARROW_INTEGRATION_GO=ON \
        ARROW_INTEGRATION_JAVA=ON \
        ARROW_INTEGRATION_JS=ON \
        ARCHERY_INTEGRATION_WITH_NANOARROW="1" \
        ARCHERY_INTEGRATION_WITH_RUST="1" \
```
Mostly preference though :)

---

ehh the whitespace didn't workout as expected in the comment editor ^^ :see_no_evil: